pkg/controller/deployment/progress_test.go

```diff
@@ -330,7 +338,7 @@ func TestSyncRolloutStatus(t *testing.T) {
 			newCond := util.GetDeploymentCondition(test.d.Status, test.conditionType)
 			switch {
 			case newCond == nil:
-				if test.d.Spec.ProgressDeadlineSeconds != nil {
+				if test.d.Spec.ProgressDeadlineSeconds != nil && *test.d.Spec.ProgressDeadlineSeconds != math.MaxInt32 {
```

Comment:
Intentionally write it explicitly in the test, so that if we break `deploymentutil.HasProgressDeadline()` in the future, this test will still catch that.